src/index.ts

```diff
@@ -0,0 +1,206 @@
+      const memberships = membershipsResponse as ItemMembership[];
+      console.log(memberships)
+
+      // get members of the item
+      const tasks = memberships.map((membership) => memberTaskManager.createGetTask(member, membership.memberId));
+      const membersResponse = await runner.runMultiple(tasks, log);
+      const members = membersResponse as Member[];
+      console.log(members)
+
+      const numActionsRetrieved = actions.length;
+      const metadata = {
+        numActionsRetrieved: numActionsRetrieved,
+        requestedSampleSize: parseInt(requestedSampleSize)
+      }
+
+      // generate responseData with actions, members, item, and metadata
+      const responseData: Analytics = new BaseAnalytics(actions, members, item, metadata);
+      
+      reply
+      .send( responseData )
+    },
+  );
+  
+  // delete all the actions matching the given `memberId`
+  fastify.delete<{ Params: IdParam }>(
+    '/members/:id/delete-actions',
+    { schema: getOneDeleteActions },
+    async ({ member, params: { id }, log,  }, reply) => {
+      const task = new DeleteActionsTask(member, id, actionService);
+      await runner.runSingle(task);
+
+      reply
+      .send( task.result )      
+    },
+  );
+
+};
+
+
+// save action (except create and delete) for the itemId with its actionType and view
+const createActionTask = async function (member: Member, itemId: string, actionType: string, view: string, geo, runner, itemTaskManager) {
+  // get item
+  const t1 = itemTaskManager.createGetTaskSequence(member, itemId);
+  const item = await runner.runSingleSequence(t1);
+  // save action
+  const extra = {memberId: member.id, itemId: item.id};
+  let action: Action = new BaseAction(member.id, item.id, member.type, item.type, actionType, view, geo, extra);
+  const t2 = new CreateActionTask(member, action, actionService);
+  await runner.runSingle(t2);
+};
+
+// function called each time there is a request in the items in graasp (onResponse hook in graasp)
+// identify and check the correct endpoint of the request
+export const createAction = async function (request, reply, runner, itemTaskManager) {
+  // check that the request is ok
+  if (correctStatusCodes.includes(reply.statusCode)) {
+    const url: string = request.url;
+    const method: string = request.method;
+    const member: Member = request.member;
+    const paramItemId: string = request.params.id;
+    const queryItemsId: string[] = request.query.id;
+    const geo = geoip.lookup(request.ip);
+
+    var view = null;
+    const {hostname} = request;
+    switch(hostname) {
+      case VIEWS.BUILDER.hostname:
+        view = VIEWS.BUILDER.name;
+        break;
+      case hostname == VIEWS.PLAYER.hostname:
+        view = VIEWS.PLAYER.name;
+        break;
+      case hostname == VIEWS.EXPLORER.hostname:
+        view = VIEWS.EXPLORER.name;
+        break;
+      default:
+        view = VIEWS.UNKNOWN.name;
+        break;
+    }
```

Comment:
Do you think (more or less correct code)
```view = VIEWS.find(v => v===hostname)?.name ?? VIEWS.UNKNOWN.name```
would be more comprehensible?

Reply:
I have tried something similar to that, but I think it is not feasible because the constants are now defined in an `enum` (it is not an array). Therefore, I think the best option now is to leave the `switch case`.